### PR TITLE
Regularization

### PR DESCRIPTION
**Pull request by azzeddinetiba**
Added Tikhonov regularization to deal with ill-conditioned input matrices, as mentioned in #184 and #107 .
The regularization is implemented as a filter on the SVD reconstruction of **X** , the filter being sigma^2 / (sigma^2 + lambda), with lambda = ```tikhonov```*||**X**||.

1. Tests passed on Ubuntu 18.04, Python 3.8
2. I added a "tutorial11", it is not a tutorial per se, but it shows the ffect of the regularization on fluid simulation data. The .npy data file is just above 100Mo though, I can't push it. It shows this effect on DMD modes only, I did see the effect on instabilites of the DMD prediction on other cases.
3. Should the bad conditioning warning (92b33ee3e87b1483bc36ef0bd70664393cff1f6e) be ommited in case regularization is wanted ?

### CODE REVIEW

**Comment by ndem0** [comment]
Dear @azzeddinetiba, first of all many thanks for the nice new feature and nice tutorial!
Before merging it, I have just few questions/requests:
- maybe we can use `tikhonov_regularization` instead of just `tikhonov` for the attribute name? In this way also people not really expert in the field can easily exploit such attribute for ill-conditioned data (with only `tikhonov` it looks quite criptic to me);
- for being more "pythonic", probably to disable the regularization it is better to have the `tikhonov` parameter to `None` instad of 0
- Can you solve the static issues that Codacy have spotted? https://app.codacy.com/gh/mathLab/PyDMD/pullRequest?prid=8712068
- Regarding the bad condition warning, we have planned to better organize the output of PyDMD, e.g with the logging module. So I would say we can skip manual print for the moment.

**Comment by azzeddinetiba** [comment]
Thank you @ndem0 for responding
 
It should be all good now.

**Comment by ndem0** [comment]
Awesome!
Thanks again, once the tests will be concluded I'll merge your additions.

File: pydmd/cdmd.py [hunk 1]
```diff
@@ -37,13 +37,19 @@ class CDMDOperator(DMDOperator):
         magnitude if `sorted_eigs='abs'`, by real part (and then by imaginary
         part to break ties) if `sorted_eigs='real'`. Default: False.
     :type sorted_eigs: {'real', 'abs'} or False
+    :param tikhonov: tikhonov parameter for regularization
+        If None, no regularization is applied, if float, it is used as the
+        lambda tikhonov parameter
```
Comment (fAndreuzzi) [inline pydmd/cdmd.py:42]:
```suggestion
    :param tikhonov: Tikhonov parameter for the regularization.
        If `None`, no regularization is applied, if `float`, it is used as the
        :math:`\lambda` tikhonov parameter.
```

**Comment by azzeddinetiba** [comment]
I also forgot to update the tutorial to tkhonov = None instead of 0. Sorry about that. I'll modify it asap.

**Comment by fAndreuzzi** [comment]
> I also forgot to update the tutorial to tkhonov = None instead of 0. Sorry about that. I'll modify it asap.

Thanks!

**Comment by azzeddinetiba** [comment]
Docstring and tutorial updated.

**Comment by azzeddinetiba** [comment]
Last commit for a docstring occurence I missed.

**Comment by fAndreuzzi** [comment]
Thank you, I was about to write about the missing update to the docstring in `cdmd.py`. 

One more question, can you explain why you decided not to include the new parameter `tikhonov_regularization` in all the DMD subclasses? I see that you modified only `cdmd.py`, `dmd.py`, `optdmd.py`, `dmdc.py`.

**Comment by azzeddinetiba** [comment]
@fAndreuzzi Hello back. I used this regularization previously on DMD and DMD with control only. I am not familiar with all the other variations. Although I supect they all use the same SVD-based dmd operator, I've thought to let it to the according authors to decide whether to include the regularization or not, in case these variations treat this problem using some other particular strategies.

**Pull request merged**